banzai/utils/image_utils.py

```diff
@@ -18,12 +27,17 @@ def get_reduction_level(header):
     return header.get('RLEVEL', '00')
 
 
-def select_images(image_list, context, image_type):
+def select_images(image_list, image_type, db_address, ignore_schedulability):
     images = []
     for filename in image_list:
         try:
             header = get_primary_header(filename)
-            if context.image_can_be_processed(header) and (image_type is None or get_obstype(header) == image_type):
+            should_process = image_can_be_processed(header, db_address)
+            should_process &= (image_type is None or get_obstype(header) == image_type)
+            if not ignore_schedulability:
+                instrument = dbs.get_instrument(header, db_address=db_address)
+                should_process &= instrument.schedulable
```

Comment:
I think there is some overlap with some of the things checked `realtime.need_to_process`, maybe some methods could be consolidated?